client/my-sites/term-tree-selector/add-term.jsx

```diff
@@ -0,0 +1,232 @@
+				textOnly: true
+			} );
+		}
+
+		if ( error !== this.state.error ) {
+			this.setState( {
+				error: error,
+				isValid: ! error
+			} );
+		}
+
+		return ! error;
+	}
+
+	validateInput( event ) {
+		if ( 13 === event.keyCode ) {
+			this.saveTerm();
+		} else {
+			this.isValid();
+		}
+	}
+
+	saveTerm() {
+		const values = this.getSelectedValues();
+		if ( ! this.isValid() ) {
+			return;
+		}
+
+		const { siteId, taxonomy } = this.props;
+
+		this.props.addTerm( siteId, taxonomy, values );
+		this.closeDialog();
+	}
+
+	render() {
+		const { labels, siteId, taxonomy, translate } = this.props;
+		const buttons = [ {
+			action: 'cancel',
+			label: translate( 'Cancel' )
+		}, {
+			action: 'add',
+			label: translate( 'Add' ),
+			isPrimary: true,
+			disabled: ! this.state.isValid,
+			onClick: this.boundSaveTerm
+		} ];
+
+		const { searchTerm, selectedParent } = this.state;
+		const query = {};
+		if ( searchTerm && searchTerm.length ) {
+			query.search = searchTerm;
+		}
+
+		const isError = this.state.error && this.state.error.length;
+
+		return (
+			<div className="term-tree-selector__add-term">
+				<Button borderless compact={ true } onClick={ this.boundOpenDialog }>
+					<Gridicon icon="folder" /> { labels.add_new_item }
+				</Button>
+				<Dialog
+					autoFocus={ false }
+					isVisible={ this.state.showDialog }
+					buttons={ buttons }
+					onClose={ this.boundCloseDialog }
+					additionalClassNames="term-tree-selector__add-term-dialog">
+					<FormSectionHeading>{ labels.add_new_item }</FormSectionHeading>
+					<FormFieldset>
+						<FormTextInput
+							autoFocus={ this.state.showDialog && ! viewport.isMobile() }
+							placeholder={ labels.new_item_name }
+							ref="termName"
+							isError={ isError }
+							onKeyUp={ this.boundValidateInput } />
+						{ isError ? <FormInputValidation isError={ true } text={ this.state.error } /> : null }
+					</FormFieldset>
+					<FormFieldset>
+						<FormLegend>
+							{ labels.parent_item }
+						</FormLegend>
+						<FormLabel>
+							<FormCheckbox ref="topLevel" checked={ this.state.isTopLevel } onChange={ this.boundOnTopLevelChange } />
+							<span>{ translate( 'Top level', { context: 'Terms: New term being created is top level' } ) }</span>
+						</FormLabel>
+						<TermTreeSelectorTerms
+							siteId={ siteId }
+							taxonomy={ taxonomy }
+							onSearch={ this.boundOnSearch }
+							onChange={ this.boundOnParentChange }
+							query={ query }
+							selected={ selectedParent }
+						/>
+					</FormFieldset>
+				</Dialog>
+			</div>
+		);
+	}
+}
+
+export default connect(
+	( state, ownProps ) => {
+		const { taxonomy } = ownProps;
+		const siteId = getSelectedSiteId( state );
+		const taxonomyDetails = getPostTypeTaxonomy( state, siteId, ownProps.postType, taxonomy ) || {};
+
+		return {
+			terms: getTerms( state, siteId, taxonomy ),
+			labels: taxonomyDetails.labels,
+			siteId
+		};
+	},
+	( dispatch ) => {
```

Comment:
You can pass a plain object as the second argument, which will have the same effect:

```js
{ addTerm }
```